scene/resources/shader.cpp

```diff
@@ -100,6 +86,20 @@ void Shader::set_code(const String &p_code) {
 		}
 	}
 
+	String type = ShaderLanguage::get_shader_type(pp_code);
```

Comment:
This might warrant a comment to make it clear why the order matters. Because otherwise it looks weird to have this between the code that gathers the `include_dependencies`, and the code that uses them. It looks like it's asking to be moved around ;)